Tools/LyTestTools/ly_test_tools/o3de/editor_test.py

```diff
@@ -56,6 +56,33 @@ class MyTestInParallel_2(EditorParallelTest):
 logger = logging.getLogger(__name__)
 
 
+def _split_batched_editor_log_file(path, log_file_to_split):
+    # text that designates the start of logging for a new test
+    test_start_split = "AutomatedTesting\\\\Gem\\\\PythonTests\\\\"
+    dir_name = os.path.dirname(path)
+
+    # the current log we are writing to
+    current_new_log_path = os.path.join(dir_name, f"startup_logging_for_{log_file_to_split}")
+    current_new_log = None
+    index = 1
+    while not current_new_log:
+        current_new_log = open(current_new_log_path + "_" + "index", "a+")
+        index += 1
+
+    # loop through the log to split, and write to the split logs
+    with open(os.path.join(dir_name, log_file_to_split)) as log_file:
+        for line in log_file:
+            split_line = line.split(test_start_split)
+            if len(split_line) > 1:
+                # found a new test case, we need to split the log
+                test_name = line.split(test_start_split)[1][:-16].replace("\\\\", "_") + ".log"
```

Comment:
Why is this split so particularly here? Could you provide an example?

Reply:
This will create a very long log file name. Testing this locally, I got the return value `'largeworlds\\dyn_veg\\EditorScripts\\LayerSpawner_InstancesPlantInAllSupportedShape.log'` A cleaner solution may be to use regex. We should also have access to all of the test module names that may be leveraged.